src/west/commands/command.py

```diff
@@ -100,6 +109,22 @@ def add_parser(self, parser_adder):
 
         return self.parser
 
+    def check_up_to_date(self):
+        '''Checks if west itself is up to date, emitting a warning if not.'''
+        try:
+            manifest = Manifest.from_file(sections=['west'])
+            west_project = manifest.west_project
```

Comment:
Suggested change to handle the only support SHA issue.

Goes together with code suggestion below.
```suggestion
            west_project = manifest.west_project
            int(west_project.revision, 16)
```